File: Autos-stock-manager-services/src/main/java/org/autostock/enums/MethodePaiement.java
package org.autostock.enums;

import com.fasterxml.jackson.annotation.JsonValue;

public enum MethodePaiement {
    COMPTANT("Comptant", "CASH"),
    CARTE("Carte bancaire", "CARD"),
    CHEQUE("Chèque", "CHEQUE"),
    VIREMENT("Virement bancaire", "TRANSFER"),
    FINANCEMENT("Financement", "LOAN");

    private final String label;   // lisible (frontend / UI)
    private final String value;   // technique (BD / API)

    MethodePaiement(String label, String value) {
        this.label = label;
        this.value = value;
    }

    public String getLabel() {
        return label;
    }

    @JsonValue  // important : utilisé lors du retour JSON
    public String getValue() {
        return value;
    }

    public static MethodePaiement fromValue(String value) {
        for (MethodePaiement m : values()) {
            if (m.value.equalsIgnoreCase(value)) {
                return m;
            }
        }
        throw new IllegalArgumentException("Methode de paiement inconnue : " + value);
    }
}

File: autos-stock-client/src/app/models/enums/MethodePaiement.ts
// Valeurs attendues par le backend : org.autostock.enums.MethodePaiement expose
// son champ `value` (et non le nom de la constante) via @JsonValue, et
// PaiementServiceImpl les resout avec fromValue(). Envoyer autre chose leve une
// IllegalArgumentException cote serveur, soit un 400.
export type MethodePaiement = 'CASH' | 'CARD' | 'CHEQUE' | 'VIREMENT' | 'FINANCEMENT';

export const METHODE_PAIEMENT_LABELS: Record<MethodePaiement, string> = {
  CASH:        'Comptant',
  CARD:        'Carte bancaire',
  CHEQUE:      'Chèque',
  VIREMENT:    'Virement bancaire',
  FINANCEMENT: 'Financement'
};

export const METHODES_PAIEMENT: { value: MethodePaiement; label: string }[] =
  (Object.keys(METHODE_PAIEMENT_LABELS) as MethodePaiement[])
    .map(value => ({value, label: METHODE_PAIEMENT_LABELS[value]}));
